### PR TITLE
Fixed bug related to settle order

### LINKED ISSUE #159: Reserving Balance is not working properly
So here's what I'm donig:
First, I deposit 500 tokens to Bob and Alice:
https://github.com/Polkadex-Substrate/polkadexTEE-worker/blob/demo-script-squashed/client/demo_openfinex.py#L201-L204
 
Then I create two orders:
 
https://github.com/Polkadex-Substrate/polkadexTEE-worker/blob/demo-script-squashed/client/demo_openfinex.py#L211-L216
 
which will be sent to openfinex, looking like:

1. ` [1,0,"admin_create_order", "","5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY","btcusd","spot","l","buy","50.0","1.0"]]`

2. `[1,1,"admin_create_order",["","5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty","btcusd","spot","l","sell","50.0","1.0"]]`
3.  We then get the trade event von openfinex:
`[5,\"on\",[\"BOB\",\"5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty\",\"btcusd\",4,\"fbda13fe-d430-11eb-9fab-0242ac120003\",\"sell\",\"d\",\"l\",\"1\",\"1\",\"0\",\"50\",\"50\",1,1624459241]]`
4. I then call into the encalve to check the balances: 
https://github.com/Polkadex-Substrate/polkadexTEE-worker/blob/demo-script-squashed/client/demo_openfinex.py#L223-L227
And they are correct! Alice has now 50 tokens of B and Bob of A and the respective lower amount of the original token

But I get these error messages:
```bash
[2021-06-23T14:40:42Z ERROR substratee_worker_enclave::polkadex_balance_storage] Unable to un-reserve balance greater than reserved balance
[2021-06-23T14:40:42Z ERROR substratee_worker_enclave::openfinex::response_handler] [Error] in polkadex gateway settle trade: NotEnoughReservedBalance
```

### PR DESCRIPTION
**Pull request by jaxter03**
There was bug in Settle trade function. It was unable to find right amount to be unreserved.

### CODE REVIEW

File: enclave/src/happy_path.rs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+use crate::polkadex_orderbook_storage::lock_storage_and_add_order;
+use crate::test_polkadex_gateway::{check_balance, create_mock_gateway};
+use polkadex_sgx_primitives::accounts::get_account;
+use polkadex_sgx_primitives::types::{
+    MarketId, Order, OrderSide, OrderType, OrderUUID, TradeEvent,
+};
+use polkadex_sgx_primitives::{AccountId, AssetId};
+use sgx_tstd::vec::Vec;
+
+pub fn test_happy_path() {
```
Comment (haerdib) [inline enclave/src/happy_path.rs:17]: This is great! Thanks alot!

File: enclave/src/polkadex_gateway.rs [hunk 2]
```diff
@@ -526,14 +521,17 @@ pub fn consume_order(
 
         (OrderType::LIMIT, OrderSide::ASK) => {
             let reserved_amount = (get_price(counter_order.price)? * counter_order.quantity) / UNIT;
+            let trade_amount = (counter_order.quantity * trade_event.price) / UNIT;
+            error!("here 1");
```
Comment (haerdib) [inline enclave/src/polkadex_gateway.rs]: I'd  suggest to remove these or change to `debug`

File: enclave/src/polkadex_gateway.rs [hunk 3]
```diff
@@ -526,14 +521,17 @@ pub fn consume_order(
...
             do_asset_exchange(&mut current_order, &mut counter_order, trade_event.amount)?;
             if counter_order.quantity > 0 {
                 polkadex_orderbook_storage::lock_storage_and_add_order(
                     counter_order.clone(),
                     maker_order_uuid,
                 )?;
             } else {
-                let amount_to_unreserve = reserved_amount - trade_event.price;
+                let amount_to_unreserve = reserved_amount - trade_amount;
+                error!("Here 2 {:?}", amount_to_unreserve);
```
Comment (haerdib) [inline enclave/src/polkadex_gateway.rs]: ^^^^

File: enclave/src/polkadex_gateway.rs [hunk 4]
```diff
@@ -542,6 +540,7 @@ pub fn consume_order(
             }
 
             if current_order.quantity > 0 {
+                error!("Here 3");
```
Comment (haerdib) [inline enclave/src/polkadex_gateway.rs]: ^^^^

File: enclave/src/tests.rs [hunk 5]
```diff
@@ -138,6 +139,7 @@ pub extern "C" fn test_main_entrance() -> size_t {
         test_proxy::test_add_proxy_account,
         test_proxy::test_remove_proxy_account,
         test_proxy::test_remove_main_account,
+		happy_path::test_happy_path,
```
Comment (haerdib) [inline enclave/src/tests.rs:142]: This should probably be moved to integration tests, not unit tests. But I'll mention in issue #152

**Review by haerdib — CHANGES_REQUESTED** [review]

File: enclave/src/tests.rs (hunk 5 above)
Comment (haerdib) [reply enclave/src/tests.rs:142]: So no need to move this for this PR.

Comment (jaxter03) [reply enclave/src/tests.rs:142]: I am sorry .. I didnt understand this ... do you want me to remove it?

Comment (haerdib) [reply enclave/src/tests.rs:142]: No, just leave it.

Comment (jaxter03) [reply enclave/src/tests.rs:142]: oh .. okay .. thanks

File: enclave/src/happy_path.rs [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+    //Add Orders into orderbook
+    let ask_limit_order_uuid: OrderUUID = (200..202).collect();
+    assert!(lock_storage_and_add_order(ask_limit_order, ask_limit_order_uuid.clone()).is_ok());
+
+    let buy_limit_order_uuid: OrderUUID = (202..204).collect();
+    assert!(lock_storage_and_add_order(buy_limit_order, buy_limit_order_uuid.clone()).is_ok());
+
+    //Order Event
+    let order_event = TradeEvent {
```
Comment (Gauthamastro) [inline enclave/src/happy_path.rs]: Rename variable to trade_event

File: enclave/src/happy_path.rs [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+        .place_order(alice.clone(), None, ask_limit_order.clone())
+        .is_ok());
+
+    // Place Bid Limit Order
+    assert!(gateway
+        .place_order(bob.clone(), None, buy_limit_order.clone())
+        .is_ok());
+
+    //Add Orders into orderbook
+    let ask_limit_order_uuid: OrderUUID = (200..202).collect();
```
Comment (Gauthamastro) [inline enclave/src/happy_path.rs]: Replace these four lines with the process_create_order function

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Review by haerdib — APPROVED** [review]
LGTM

**Pull request merged**